src/applications/static-pages/facilities/createFacilityPage.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import { fetchFacility } from './actions';
+
+export default async function createFacilityPage(store) {
+  let facilityID = '';
+  const detailWidget = document.querySelector(
+    `[data-widget-type="facility-detail"]`,
+  );
+  const mapWidget = document.querySelector(`[data-widget-type="facility-map"]`);
+  const scoreWidget = document.querySelector(
+    `[data-widget-type="facility-patient-satisfaction-scores"]`,
+  );
+  const waitTimeWidgets = Array.from(
+    document.querySelectorAll(
+      `[data-widget-type="facility-appointment-wait-times-widget"]`,
+    ),
+  );
+
+  if (detailWidget) {
+    facilityID = JSON.parse(detailWidget.dataset.facility);
+  }
+  if (mapWidget && !facilityID) {
+    facilityID = JSON.parse(mapWidget.dataset.facility);
+  }
+  if (scoreWidget && !facilityID) {
+    facilityID = JSON.parse(scoreWidget.dataset.facility);
+  }
+  if (waitTimeWidgets.length && !facilityID) {
+    facilityID = JSON.parse(waitTimeWidgets[0].dataset.facility);
+  }
+  if (facilityID) {
+    store.dispatch(fetchFacility(facilityID));
+    if (mapWidget) {
+      const {
+        default: FacilityMapWidget,
+      } = await import(/* webpackChunkName: "facility-detail" */ './FacilityMapWidget');
+      ReactDOM.render(
+        <Provider store={store}>
+          <FacilityMapWidget />
+        </Provider>,
+        mapWidget,
+      );
+    }
+    if (detailWidget) {
+      const {
+        default: FacilityDetailWidget,
+      } = await import(/* webpackChunkName: "facility-detail" */ './FacilityDetailWidget');
+      ReactDOM.render(
+        <Provider store={store}>
+          <FacilityDetailWidget />
+        </Provider>,
+        detailWidget,
+      );
+    }
+    if (scoreWidget) {
+      const {
+        default: FacilityPatientSatisfactionScoresWidget,
+      } = await import(/* webpackChunkName: "facility-detail" */ './FacilityPatientSatisfactionScoresWidget');
+      ReactDOM.render(
+        <Provider store={store}>
+          <FacilityPatientSatisfactionScoresWidget />
+        </Provider>,
+        scoreWidget,
+      );
+    }
+    if (waitTimeWidgets.length) {
+      const {
+        default: FacilityAppointmentWaitTimesWidget,
+      } = await import(/* webpackChunkName: "facility-patient-satisfaction-scores" */ './FacilityAppointmentWaitTimesWidget');
```

Comment:
Is this on a different page than the detail and score widgets? There might be some benefit in making the webpack chunkname the same for all of them if it's the same page or if we expect people to usually go to pages where both are used, in one session.

Reply:
Same page - changed :+1: